GitExtUtils/CredentialManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Net;
+
+namespace GitExtUtils
+{
+    public static class CredentialManager
+    {
+        private const string TargetPrefix = "GitExtensions_";
+
+        private static string GetTarget(string rawTarget)
```

Comment:
All public API must validate input parameters. By not doing this `rawTarget` containing `null` or whitespace falls through to here... IMO situations like that should be treated as an exception.